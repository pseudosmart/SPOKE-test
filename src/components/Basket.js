import React, {Component} from 'react'
import { connect } from 'react-redux'
import BasketItem from './BasketItem'

export class Basket extends Component {
  render() {
    const basket = this.props.basket || []
    return (
      <div className='m-3'>
        <h2>Basket</h2>
        {basket.map((basketItem, i) => {
          return <BasketItem key={basketItem.id + i} basketItem={basketItem} />
        })}
          <p className='h2'>Total: Â£{this.props.total} </p>
      </div>
    )
  }
}


export const mapStateToProps = (state) => {
  return {
    basket: state.basket.currentBasket,
    total: state.basket.total
  }
}

export default connect(mapStateToProps)(Basket)