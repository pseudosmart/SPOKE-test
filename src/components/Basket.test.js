import React from 'react'
import { shallow } from 'enzyme'
import { Basket, mapStateToProps } from './Basket'

let defaultValues = {
  basket: {
    currentBasket: [
      {
        title: 'Testing title',
        productType: 'Type test',
        price: '45.00',
        id: 1
      },
      {
        title: 'Testing title2',
        productType: 'Other type',
        price: '98.00',
        id: 2
      }
    ]
  },
  total: 300
}

test('renders correctly', () => {
  const props = {
    basket: [
      {
        title: 'Testing title',
        productType: 'Type test',
        price: '45.00',
        id: 1
      },
      {
        title: 'Testing title2',
        productType: 'Other type',
        price: '98.00',
        id: 2
      }
  ],
  total: 200.00
  }
  const wrapper = shallow(<Basket {...props} />)
  expect(wrapper).toMatchSnapshot()
})

test('mapStateToProps', () => {
  expect(mapStateToProps(defaultValues)).toMatchSnapshot()
})
