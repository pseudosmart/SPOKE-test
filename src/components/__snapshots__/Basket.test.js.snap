// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapStateToProps 1`] = `
Object {
  "basket": Array [
    Object {
      "id": 1,
      "price": "45.00",
      "productType": "Type test",
      "title": "Testing title",
    },
    Object {
      "id": 2,
      "price": "98.00",
      "productType": "Other type",
      "title": "Testing title2",
    },
  ],
  "total": undefined,
}
`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
